src/frontends/tensorflow_common/src/op/pack.cpp

```diff
@@ -16,22 +17,56 @@ namespace tensorflow {
 namespace op {
 
 OutputVector translate_pack_op(const NodeContext& node) {
-    default_op_checks(node, 1, {"Pack", "PACK"});
-    auto num_size = static_cast<int>(node.get_input_size());
+    default_op_checks(node, 1, {"Pack", "PACK"}, true);
 
+    auto num_size = static_cast<int>(node.get_input_size());
     auto axis = node.get_attribute<int64_t>("axis", 0);
     auto axis_const = make_shared<v0::Constant>(element::i64, Shape{}, axis);
 
     OutputVector concat_inputs;
+    bool has_complex_input = false;
+    ov::element::Type complex_part_type;
+
     for (int ind = 0; ind < num_size; ++ind) {
         auto in = node.get_input(ind);
+        auto complex_type_mark = as_type_ptr<ComplexTypeMark>(in.get_node_shared_ptr());
+
+        if (complex_type_mark) {
+            has_complex_input = true;
+            complex_part_type = complex_type_mark->get_complex_part_type();
+            in = complex_type_mark->input_value(0);
+        }
+
         concat_inputs.push_back(make_shared<v0::Unsqueeze>(in, axis_const));
     }
 
-    auto pack = make_shared<v0::Concat>(concat_inputs, axis);
-    set_node_name(node.get_name(), pack);
-    return {pack};
+    auto concat = make_shared<v0::Concat>(concat_inputs, axis);
+
+    if (has_complex_input) {
+        OutputVector extended_concat_inputs;
+
+        for (const auto& input : concat_inputs) {
+            auto complex_type_mark = as_type_ptr<ComplexTypeMark>(input.get_node_shared_ptr());
+            if (complex_type_mark) {
+                auto real_part = make_shared<v0::Unsqueeze>(complex_type_mark->input_value(0), axis_const);
+                extended_concat_inputs.push_back(real_part);
+                auto imag_part = make_shared<v0::Unsqueeze>(complex_type_mark->input_value(1), axis_const);
+                extended_concat_inputs.push_back(imag_part);
+            } else {
+                extended_concat_inputs.push_back(input);
+            }
+        }
```

Comment:
is it possible to concatenate complex tensors without extraction of real and imaginary parts?
are you sure that type of operands can be different: sort of mix of real and complex types?

Reply:
I extracted the real and imaginary parts because there is no support for complex types. Could you please confirm if there are alternative methods I should explore to handle complex tensors more effectively within the framework? Regarding the second question, I am considering adding a separate implementation path for mixed-type tensors, treating real inputs as having zero imaginary parts and for tensors with only real types, adding an imaginary part with zeros to maintain consistency. Would this approach align with the project's requirements?

Reply:
Any idea how to support it without extraction real and imaginary parts?
Did you check TensorFlow documentation and what it says about support of mixed types?

Reply:
In TensorFlow, when a real tensor and complex type tensor are stacked (as the newer version of Tensorflow changed the name of the pack operation to stack), an InvalidArgumentError is thrown. So the workaround was to convert the real tensor into a complex one to stack it together. A mixed tensor is often considered a complex type for ease as a TypeError will be thrown otherwise. I will refer to the Openvino operations which support complex types to see how it is handled there. Thank you.